src/openms/source/CHEMISTRY/ModificationsDB.cpp

```diff
@@ -574,43 +572,45 @@ namespace OpenMS
     // now use the term and all synonyms to build the database
     for (multimap<String, ResidueModification>::const_iterator it = all_mods.begin(); it != all_mods.end(); ++it)
     {
-
-      // check whether a unimod definition already exists, then simply add synonyms to it
-      if (it->second.getUniModRecordId() > 0)
+      #pragma omp critical(OpenMS_ModificationsDB)
```

Comment:
this may actually slow down execution if you acquire / release the lock in each loop - why not just put the lock on the top-level?

Reply:
Good point. Will do